### ISSUE TITLE
google.protobuf.empty is not available in protobuf-lite

### ISSUE THREAD

**Issue by johndanenRV** [rossvideo/Catena#290]
Building catena with model:"LITE" and connection:"REST" causes an error when linking param.pb.cc. This is because param.proto uses google.protobuf.empty which is not available in the protobuf-lite library. This issue did not appear when using gRPC connection because the gRPC library is linked to the full protobuf library.

The easiest fix for this is probably to define our own empty message instead of using the one provided by protobuf. This would require changing the namespace of the empty message in the C++ sdk and might also break the Java sdk.

**Comment by johndanenRV** [comment]
I was curious to see how large of a difference it would make to just use the full protobuf library, so I did some test. All of these tests were done with gRPC disabled.

`proto_optimization = LITE_RUNTIME`, `empty.proto` commented out, linked to `libprotobuf-lite`: example/start_here Size: 15M

`proto_optimization = LITE_RUNTIME`, `empty.proto` commented out, linked to full `libprotobuf`: example/start_here Size: 15M

`proto_optimization = LITE_RUNTIME`, `empty.proto` uncommented, linked to full `libprotobuf`: example/start_here Size: 23M

Needless to say, increasing the file size by 8M just to define an empty message seems a bit excessive. I then decided to do the same test with gRPC enabled.

using `google.protobuf.empty`: example/status_update Size: 75M

using locally defined `empty`: example/status_update Size: 67M

Again, we save 8M by not using `google.protobuf.empty`. The good news here is that despite gRPC being linked against the full libprotobuf, the file size only increases if we are using the full protobuf library.
